json/src/main/java/com/basistech/rosette/dm/jackson/VersionProperty.java

```diff
@@ -0,0 +1,63 @@
+/*
+* Copyright 2016 Basis Technology Corp.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package com.basistech.rosette.dm.jackson;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.cfg.MapperConfig;
+import com.fasterxml.jackson.databind.introspect.AnnotatedClass;
+import com.fasterxml.jackson.databind.introspect.BeanPropertyDefinition;
+import com.fasterxml.jackson.databind.ser.VirtualBeanPropertyWriter;
+import com.fasterxml.jackson.databind.util.Annotations;
+
+/**
+ * Custom injected property that injects the version of the ADM _as a Json data model_.
+ * Version history:
+ * <ul>
+ *     <li>1.0.0 -- the 'classic' ADM</li>
```

Comment:
I added "version": "2.0.0" to a json adm file and tried to
deserialize.  I got an exception, which is expected.  But the message
is a bit funny:

```
com.fasterxml.jackson.databind.exc.InvalidFormatException:
  Can not construct instance of java.lang.String from String value '[N/A]': Incompatible ADM version 2.0.0
```

"Incompatible ADM version 2.0.0" is fine, though might be better to
emit the acceptable version range.

But the text about not constructing String and "[N/A]" is not helpful
here.

If I use version 1.2.0, no exception is thrown.  Is this intended?